ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -12,204 +12,188 @@
-            <TextField
-              type="number"
-              label="Buffer width"
-              min={0.000001}
-              value={`${props[widthKey]}`}
-              onChange={cursor(widthKey)}
-            />
-          </Units>
-        </div>
-        {this.state.isValid ? (
-          ''
-        ) : (
-          <Invalid>
-            &nbsp;
-            <span className="fa fa-exclamation-triangle" />
-            &nbsp; {this.invalidMessage} &nbsp; &nbsp;
-            <span className="fa fa-times" onClick={this.removeErrorBox} />
-          </Invalid>
-        )}
-      </React.Fragment>
-    )
+  return value
+}
+
+function convertWkt(value, numCoords) {
+  const coordinatePairs = value.match(coordinatePairRegex)
+  if (!coordinatePairs || coordinatePairs.length < numCoords) {
+    return value
   }
-  removeErrorBox() {
-    this.setState({ isValid: true })
+  const coordinates = coordinatePairs.map(coord => coord.replace(' ', ','))
+  return buildWktString(coordinates)
+}
+
+function convertMultiWkt(isPolygon, value) {
+  if (isPolygon && !value.endsWith(')))')) {
+    return value
+  } else if (!value.endsWith('))')) {
+    return value
   }
-  isValidInput(value) {
-    this.invalidMessage = ''
-    this.setState({ value, isValid: this.isValidPolygon(value) })
+  const splitter = isPolygon ? '))' : ')'
+  const numPoints = isPolygon ? 4 : 2
+  let shapes = value
+    .split(splitter)
+    .map(shape => shape.match(coordinatePairRegex))
+  shapes = shapes
+    .filter(shape => shape !== null && shape.length >= numPoints)
+    .map(shape => shape.map(coordinatePair => coordinatePair.replace(' ', ',')))
+  return shapes.length === 0
+    ? value
+    : shapes.length === 1
+      ? this.buildWktString(shapes[0])
+      : '[' + shapes.map(shapeCoords => this.buildWktString(shapeCoords)) + ']'
+}
+
+function is2DArray(coordinates) {
+  try {
+    const parsedCoords = JSON.parse(coordinates)
+    return Array.isArray(parsedCoords) && Array.isArray(parsedCoords[0])
+  } catch (e) {
+    return false
   }
-  is2DArray(coordinates) {
-    try {
-      const parsedCoords = JSON.parse(coordinates)
-      return Array.isArray(parsedCoords) && Array.isArray(parsedCoords[0])
-    } catch (e) {
-      return false
-    }
+}
+
+function hasPointError(point) {
+  if (
+    point.length !== 2 ||
+    (Number.isNaN(Number.parseFloat(point[0])) &&
+      Number.isNaN(Number.parseFloat(point[1])))
+  ) {
+    return true
+  } else if (
+    point[0] > 180 ||
+    point[0] < -180 ||
+    point[1] > 90 ||
+    point[1] < -90
+  ) {
+    return true
   }
-  validatePoint(point) {
+  return false
+}
+
+const BaseLine = props => {
+  const { label, geometryKey, setState, unitKey, widthKey, mode } = props
+  const [currentValue, setCurrentValue] = useState(
+    JSON.stringify(props[geometryKey])
+  )
+  const [baseLineError, setBaseLineError] = useState({
+    error: false,
+    message: '',
+  })
```

Comment:
I'd suggest pulling the initial state into a const here too since it's in two places